### ISSUE TITLE
sort-comp not working by cli

### ISSUE THREAD

**Issue by FezVrasta** [yannickcr/eslint-plugin-react#653]
I have the `sort-comp` rule enabled, inside Atom (with ESlint package) everything works.

When I run the linter from CLI instead, the linter doesn't report any problem.

```
$(npm bin)/eslint . --ignore-path .gitignore
```

```
# .gitignore file
/coverage
/build
/node_modules
.happypack
.tmp
.DS_Store
npm-debug.log
.idea
```

For instance, I'm putting a `submit()` method after `render()`

Am I doing something wrong or what?

**Comment by ljharb** [comment]
Where's your eslint config? Typically it'd be in a `.eslintrc` file in the root of the project.

**Comment by FezVrasta** [comment]
It's in `package.json`


```
"eslintConfig": {
    "parser": "babel-eslint",
    "extends": [
      "eslint:recommended",
      "plugin:react/recommended",
      "airbnb"
    ],
    "env": {
      "browser": true,
      "node": true,
      "mocha": true,
      "es6": true
    },
    "plugins": [
      "babel",
      "react",
      "jsx-a11y"
    ],
    "parserOptions": {
      "ecmaVersion": 6,
      "sourceType": "module",
      "ecmaFeatures": {
        "jsx": true
      }
    },
    "rules": {
      "import/no-unresolved": "error",
      "no-else-return": "off",
      "comma-dangle": [
        "error",
        "always-multiline"
      ],
      "indent": [
        "error",
        2,
        {
          "SwitchCase": 1
        }
      ],
      "spaced-comment": [
        "error",
        "always",
        {
          "markers": [
            "#"
          ]
        }
      ],
      "no-param-reassign": [
        "error",
        {
          "props": false
        }
      ],
      "new-cap": "off",
      "babel/new-cap": "warn",
      "quotes": [
        "error",
        "single",
        {
          "avoidEscape": true,
          "allowTemplateLiterals": true
        }
      ],
      "jsx-quotes": [
        "error",
        "prefer-single"
      ],
      "no-console": "warn",
      "no-debugger": "warn",
      "no-restricted-syntax": "off",
      "max-len": [
        "warn",
        {
          "code": 120,
          "ignoreUrls": true
        }
      ],
      "prefer-template": "error",
      "no-use-before-define": "off",
      "newline-per-chained-call": "off",
      "arrow-body-style": [
        "error",
        "as-needed"
      ],
      "react/jsx-indent": [
        "error",
        2
      ],
      "react/jsx-indent-props": [
        "error",
        2
      ],
      "react/jsx-first-prop-new-line": [
        "error",
        "multiline"
      ],
      "react/prefer-stateless-function": "off",
      "jsx-a11y/href-no-hash": "error",
      "jsx-a11y/label-has-for": "error",
      "jsx-a11y/mouse-events-have-key-events": "error",
      "jsx-a11y/role-has-required-aria-props": "error",
      "jsx-a11y/role-supports-aria-props": "error",
      "jsx-a11y/aria-props": "error"
    },
    "settings": {
      "import/resolver": {
        "webpack": {
          "config": "./internals/webpack/webpack.test.babel.js"
        }
      }
    }
  },
```

**Comment by ljharb** [comment]
What does `npm ls` say about all the eslint-related things?

**Comment by FezVrasta** [comment]
```
react$ npm ls | grep eslint
├─┬ babel-eslint@6.0.4
├─┬ eslint@2.12.0
├─┬ eslint-config-airbnb@9.0.1
│ └── eslint-config-airbnb-base@3.0.1
├─┬ eslint-import-resolver-webpack@0.3.0
├── eslint-plugin-babel@3.2.0
├─┬ eslint-plugin-import@1.8.1
│ ├── eslint-import-resolver-node@0.2.0
├─┬ eslint-plugin-jsx-a11y@1.5.3
├── eslint-plugin-react@5.2.1
```

**Comment by ljharb** [comment]
@FezVrasta just to confirm, if you try with `eslint-plugin-react@5.2.2` and `eslint@2.13.1`?

Could you also share the code of the component that errors in your editor but not on the CLI?

**Comment by FezVrasta** [comment]
Yes it happens even with the versions you have suggested.

The component is:

```
/**
 *
 * Button
 *
 */

import React, { PropTypes } from 'react';
import styleguide from './styleguide';
import CSSModules from 'react-css-modules';
import styles from './styles.styl';

@CSSModules(styles, { allowMultiple: true })
class Button extends React.Component {
  static styleguide = styleguide;

  static defaultProps = {
    type: 'button',
    primary: false,
    disabled: false,
  };

  render() {
    // Type of the button
    const type = (this.props.type === 'anchor') ? 'button' : this.props.type;

    // Style of Button's inner element
    let innerStyleName = 'Button__inner';
    if (this.props.primary) innerStyleName += ' Button__inner--primary';

    if (this.props.type === 'anchor') {
      innerStyleName += ' Button__inner--anchor';
    } else {
      innerStyleName += ' Button__inner--button';
    }

    if (this.props.disabled) {
      innerStyleName += ' Button__inner--disabled';
    }

    return (
      <button
        type={type}
        styleName='Button'
        className={this.props.className}
        onClick={this.props.onClick}
        style={this.props.style}
        disabled={this.props.disabled}
      >
        <div
          styleName={innerStyleName}
          style={{ backgroundColor: this.props.color }}
          className={this.props.innerClassName}
        >
          {this.props.children}
        </div>
      </button>
    );
  }

  submit() {
    return true;
  }
}

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit', 'reset', 'anchor']),
  primary: PropTypes.bool,
  onClick: PropTypes.func,
  className: PropTypes.string,
  innerClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
  style: PropTypes.object,
  disabled: PropTypes.bool,
  color: PropTypes.string,
};

export default Button;
```

**Comment by ljharb** [comment]
If you remove/comment out the static class properties, does the bug still occur?

**Comment by yannickcr** [comment]
Hi, I do not see the `react/sort-comp` rule in your ESLint configuration. Are you sure it is correctly enabled?

**Comment by FezVrasta** [comment]
@ljharb If I remove the static properties nothing changes
@yannickcr it's defined inside one of the defaults I import

**Comment by TSMMark** [comment]
Sort-comp does not work for me either. Opened #1490

**Comment by ljharb** [comment]
We have lots of tests, and this is a very old issue; I'm going to close this in favor of #1490. Please file a new issue if you're still seeing problems with the latest version of the plugin.
